doc/dataset_filtering.rdoc

```diff
@@ -106,6 +106,21 @@ You can also use placeholders with :placeholder and a hash of placeholder values
   items.where(Sequel.lit('category = :category', category: "ruby")).sql
   # "SELECT * FROM items WHERE category = 'ruby'"
 
+In order to combine AND and OR together, you have a few options:
+
+  items.where(category: nil).or(category: "ruby")
+  # SELECT * FROM items WHERE (category IS NULL) OR (category = 'ruby')
+
+This won't work if you add other conditions:
+
+  items.where(name: "Programming in Ruby").where(category: nil).or(category: 'ruby')
+  # SELECT * FROM items WHERE ((name = 'Programming in Ruby') AND (category IS NULL)) OR (category = 'ruby')
+
+The OR applies globally and not locally. To fix this, use Sequel.|:
+
+  items.where(name: "Programming in Ruby").where(Sequel.|([:category, nil], [:category, "ruby"]))
```

Comment:
I think the most readable way for this kind of stuff is using sequel epressions, like:
```ruby
where((Sequel[:category] =~ nil) | Sequel[:category] =~ ruby))
```
This can also be achieved with virtual row blocks.

Reply:
Or even:

```ruby
.where(Sequel[category: nil] | Sequel[category: ruby])
```

Reply:
@AlexWayfer  Wow, that's cool stuff, didn't see that before 👍

Reply:
It doesn't work for non-equal operators, like `Sequel[:count] > 0`, but it's very handful, yeah.

Also, you can combine them all, like:

```ruby
.where(Sequel[name: "Programming in Ruby"] & (Sequel[category: nil] | Sequel[category: ruby]))
```